src/package.rs

```diff
@@ -205,12 +164,38 @@ pub struct Package {
     pub setup_hint: Option<String>,
 }
 
+async fn new_zone_archive_builder(
+    package_name: &str,
+    output_directory: &Path,
+) -> Result<tar::Builder<flate2::write::GzEncoder<std::fs::File>>> {
+    let tarfile = output_directory.join(format!("{}.tar.gz", package_name));
+    let file = create_tarfile(tarfile)?;
+    // TODO: Consider using async compression, async tar.
+    // It's not the *worst* thing in the world for a packaging tool to block
+    // here, but it would help the other async threads remain responsive if
+    // we avoided blocking.
```

Comment:
I realize this comment has been moved and not added in this PR, but just wondering out loud - does all of the potentially-blocking compression and I/O happen just within `.append_file()`? Could we address this TODO with a wrapper around `tar::Builder` that had an async version of `.append_file()` via one of the tokio::task::*block* helpers?

Not suggesting this happen as part of this PR though!

Reply:
I went ahead and added a trait, `AsyncAppendFIle` which makes the heavy-lifting operations sync by using `block_in_place`.